src/frontend/src/common/components/library/data_subview/index.tsx

```diff
@@ -44,46 +64,53 @@ function searchReducer(state: SearchState, action: SearchState): SearchState {
 }
 
 function tsvDataMap(
-  tableData: TableItem[],
+  checkedSamples: string[],
+  tableData: TableItem[] | undefined,
   headers: Header[],
   subheaders: Record<string, SubHeader[]>
-): [string[], string[][]] {
+): [string[], string[][]] | undefined {
   const headersDownload = [...headers];
   headersDownload[7] = {
     key: "CZBFailedGenomeRecovery",
     sortKey: ["CZBFailedGenomeRecovery"],
     text: "Genome Recovery",
   };
-  const tsvData = tableData.map((entry) => {
-    return headersDownload.flatMap((header) => {
-      if (
-        typeof entry[header.key] === "object" &&
-        Object.prototype.hasOwnProperty.call(subheaders, header.key)
-      ) {
-        const subEntry = entry[header.key] as Record<string, JSONPrimitive>;
-        return subheaders[header.key].map((subheader) =>
-          String(subEntry[subheader.key])
-        );
-      }
-      if (header.key == "CZBFailedGenomeRecovery") {
-        if (entry[header.key]) {
-          return "Failed";
+  if (tableData !== undefined) {
```

Comment:
we can shorten this with just `if (tableData) { ... }`

Reply:
that does seem like a better idea 😆